module/jobqueue/component_consumer.go

```diff
@@ -0,0 +1,146 @@
+package jobqueue
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/storage"
+)
+
+type ComponentConsumer struct {
+	component.Component
+
+	cm           *component.ComponentManager
+	consumer     module.JobConsumer
+	jobs         module.Jobs
+	workSignal   <-chan struct{}
+	preNotifier  NotifyDone
+	postNotifier NotifyDone
+	log          zerolog.Logger
+}
+
+// NewComponentConsumer creates a new ComponentConsumer consumer
+func NewComponentConsumer(
+	log zerolog.Logger,
+	workSignal <-chan struct{},
+	progress storage.ConsumerProgress,
+	jobs module.Jobs,
+	defaultIndex uint64,
+	processor JobProcessor, // method used to process jobs
+	maxProcessing uint64,
+	maxSearchAhead uint64,
+) *ComponentConsumer {
+
+	c := &ComponentConsumer{
+		workSignal: workSignal,
+		jobs:       jobs,
+		log:        log,
+	}
+
+	// create a worker pool with maxProcessing workers to process jobs
+	worker := NewWorkerPool(
+		processor,
+		func(id module.JobID) { c.NotifyJobIsDone(id) },
+		maxProcessing,
+	)
+	c.consumer = NewConsumer(c.log, c.jobs, progress, worker, maxProcessing, maxSearchAhead)
+
+	builder := component.NewComponentManagerBuilder().
+		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			worker.Start(ctx)
+			<-worker.Ready()
```

Comment:
we should use `util.WaitReady` here to also check for ctx cancelled

Reply:
good catch